database_talker/scripts/hoge.py

```diff
@@ -342,20 +375,37 @@ def cb(self, msg):
             rospy.logwarn("received dialogflow action '{}'".format(result.response.action))
             print(result.response)
             if result.response.action == 'input.unknown':
-                self.publish_google_chat_card("🤖")
+                self.publish_google_chat_card("🤖", space)
             elif result.response.action == 'make_reply':
+                self.publish_google_chat_card("・・・", space)
+
+                parameters = yaml.safe_load(result.response.parameters)
+                startdate=datetime.datetime.now(JST)-datetime.timedelta(hours=24)
+                duration=datetime.timedelta(hours=24)
+                if parameters['date']:
+                    startdate = datetime.datetime.strptime(re.sub('\+(\d+):(\d+)$', '+\\1\\2',parameters['date']), "%Y-%m-%dT%H:%M:%S%z")
+                    duration = datetime.timedelta(hours=24)
+                if parameters['date-period']:
+                    startdate = datetime.datetime.strptime(re.sub('\+(\d+):(\d+)$', '+\\1\\2',parameters['date-period']['startDate']), "%Y-%m-%dT%H:%M:%S%z")
+                    duration = datetime.datetime.strptime(re.sub('\+(\d+):(\d+)$', '+\\1\\2',parameters['date-period']['endDate']), "%Y-%m-%dT%H:%M:%S%z") - startdate
```

Comment:
I doubt that this section cause something wrong with searching database

Reply:
Sorry, I misunderstood the problem. This section has no problem.

This is because a long search, such as a 24-hour search, will not return results until the search is completed, unless the number of data matching the condition is equal to the number of limit items.